eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Ratify.hs

```diff
@@ -184,35 +195,83 @@ dRepAcceptedRatio RatifyEnv {reDRepDistr, reDRepState, reCurrentEpoch} gasDRepVo
 
     (yesStake, totalExcludingAbstainStake) = Map.foldlWithKey' accumStake (0, 0) reDRepDistr
 
+verifyPrev ::
+  forall era.
+  GovAction era ->
+  EnactState era ->
+  Bool
+verifyPrev act ens =
+  case act of
+    NoConfidence prev -> check ensCommitteeHPL prev
+    NewCommittee prev _ _ -> check ensCommitteeHPL prev
+    NewConstitution prev _ -> check ensConstitutionHPL prev
+    HardForkInitiation prev _ -> check ensProtVerHPL prev
+    ParameterChange prev _ -> check ensCurPParamsHPL prev
+    TreasuryWithdrawals {} -> True
+    InfoAction {} -> True
+  where
+    check ::
+      Lens' (EnactState era) (HashProtected r a (EraCrypto era)) ->
+      StrictMaybe (PrevGovActionId r (EraCrypto era)) ->
+      Bool
+    check h prev = hpId (ens ^. h) == prev
+
+delayed ::
+  GovAction era ->
+  EnactState era ->
+  Bool ->
+  Bool
+delayed a es d = d || not (verifyPrev a es) || not (withdrawalCanWithdraw a)
```

Comment:
There is too much negation going on here, let's not do that. It is much cleaner the way it is right now were we check the conjunction of all checks that a required for acceptance, rather than everything is needed for delaying.